src/components/input/input.tsx

```diff
@@ -0,0 +1,73 @@
+import {
+  type ComponentPropsWithoutRef,
+  type ReactElement,
+  forwardRef,
+  useId,
+  useState
+} from "react"
+
+import clsx from "clsx"
+
+import s from "./input.module.scss"
+
+import eyeIcon from "../../assets/icons/eye.svg"
+
+type Props = {
+  error?: string
+  labelText?: ReactElement
+  type?: "email" | "password" | "search" | "text"
+} & ComponentPropsWithoutRef<"input">
+
+export const Input = forwardRef<HTMLInputElement, Props>((props, ref) => {
+  const { className, error, id, labelText, type = "text", ...restProps } = props
+
+  const generatedId = useId()
+  const inputId = id ?? generatedId
+  const isInputPassword = type === "password"
+  const isInputSearch = type === "search"
+
+  const [isPasswordHidden, setPasswordHidden] = useState(isInputPassword)
+  const toggleHidePassword = () => setPasswordHidden(!isPasswordHidden)
+  const hidePasswordTitle = isPasswordHidden ? "Показать пароль" : "Скрыть пароль"
+
+  let inputType
+
+  if (isInputPassword) {
+    inputType = isPasswordHidden ? "password" : "text"
+  } else {
+    inputType = type
+  }
```

Comment:
```
const isInputPassword = type === "password"
const isInputSearch = type === "search"

const hidePasswordTitle = isPasswordHidden ? "Показать пароль" : "Скрыть пароль"

let inputType

  if (isInputPassword) {
    inputType = isPasswordHidden ? "password" : "text"
  } else {
    inputType = type
  }
```

It can be a function

Reply:
I don't get it. Can I have an example, please? 🙏